packages/provider-utils/src/zod-to-json-schema/README.md

```diff
@@ -0,0 +1,5 @@
+# `zod-to-json-schema`
+
+Originally copied from https://github.com/StefanTerdell/zod-to-json-schema because its `peerDependency` on `"zod": "^3.24.1"` while `ai` needs to support `zod@4` if users already use it. We want to avoid having both `zod@3` and `zod@4` in the dependency tree of our users.
+
+The original code was released under [ISC license](https://github.com/StefanTerdell/zod-to-json-schema/blob/3893d8565e6a2278d3b7b9c59514a1db2d4fe861/LICENSE).
```

Comment:
I'm no legal expert, but I think we have to copy the LICENSE file 1:1 to adhere to the "provided that the above copyright notice and this permission notice appear in all copies." clause. We can then link to it from here.